deploy/Chart/templates/rp/deployment.yaml

```diff
@@ -69,6 +69,10 @@ spec:
         volumeMounts:
         - name: config-volume
           mountPath: /etc/config
+        {{- if eq .Values.global.aws.irsa.enabled true }}
+        - name: aws-iam-token
+          mountPath: /var/run/secrets/eks.amazonaws.com/serviceaccount
```

Comment:
@nithyatsu did we say during design discussion that IRSA can be enabled on non EKS clusters as well? Would this mount path still be applicable on other clusters?

Reply:
Hi Karishma, IRSA is not specific to eks clusters. But the first iteration of the feature is based on eks. 
We would have to make the token path configurable in order to support it on other clusters too. Would it be OK to log a ticket to cover this?

Reply:
yeah that sounds good as long as it is intentional and being tracked

Reply:
created https://github.com/radius-project/radius/issues/7774 for tracking this.

Reply:
Is this actually something we'd have to change for another cluster?

From what I understand, the AWS SDKS will use the value of [AWS_WEB_IDENTITY_TOKEN_FILE](https://docs.aws.amazon.com/cli/v1/userguide/cli-configure-envvars.html) to load the token. There's nothing special about this file path (or any other file path) as long as the environment variable points to the token on disk.

Reply:
In code, we added the token file location as a const: https://github.com/radius-project/radius/blob/007f7e68fea64a93103c18274aeefaacd3a6a217/pkg/recipes/terraform/config/providers/aws.go#L56

The mount path should work, but it felt odd to reference eks in path for a cluster that was not on eks. Or is it OK since IRSA is a AWS feature? Also, instead of const, should I have looked to set the env variable for the pods?

Reply:
IMO we should do the following:

- Set the env-var `AWS_WEB_IDENTITY_TOKEN_FILE` to `/var/run/secrets/eks.amazonaws.com/serviceaccount/token` in the helm chart when IRSA is enabled. This is also done by the webhook [here](https://github.com/aws/amazon-eks-pod-identity-webhook/blob/bd3e26dba0be6d8f8ac26578defa94582b026324/pkg/handler/handler.go#L244).
- We can simplify the logic for Terraform because Terraform also uses `AWS_WEB_IDENTITY_TOKEN_FILE`. [link](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#container-credentials)

Then it doesn't matter what the path is, because it's not hardcoded.

Reply:
updated the ticket https://github.com/radius-project/radius/issues/7774 to implement this comment.